File: 2022/haskell/app/Day1.hs
module Day1 (day1) where

import System.IO
import Data.List
import Data.List.Split

fp :: FilePath
fp = "../data/Day1.txt"


day1 :: IO ()
day1 = do
    contents <- readFile fp
    let cals = splitOn [""] (lines contents)
    let sortedCals = sortOn negate $ map (sum . map read) cals :: [Int]
    putStrLn "Day 1:"
    putStrLn $ "Part 1: " ++ show (head sortedCals)
    putStrLn $ "Part 2: " ++ show (sum $ take 3 sortedCals)

File: 2022/rust/src/day1.rs
static FILE_PATH: &'static str = "../data/Day1.txt";

pub fn day1() {
    let contents = std::fs::read_to_string(FILE_PATH).expect("Broken file");
    let mut summations: Vec<i32> = contents
        .split("\n\n")
        .map(|x| {
            x.split("\n")
                .map(|x| x.parse::<i32>().unwrap_or(0)).sum()
        }).collect();
    summations.sort_by_key(|x| -x);
    println!("Day 1:");
    println!("Part 1: {:?}", summations[0]);
    println!("Part 2: {:?}", summations[..2].iter().sum::<i32>());
}
